Shared/sdk/CVector.h

```diff
@@ -26,53 +26,64 @@
 class CVector
 {
 public:
-    float fX, fY, fZ;
+    float fX = 0.0f;
+    float fY = 0.0f;
+    float fZ = 0.0f;
 
-    CVector()
+    constexpr CVector() = default;
+
+    constexpr CVector(float x, float y, float z) :
+        fX(x),
+        fY(y),
+        fZ(z)
     {
-        this->fX = 0;
-        this->fY = 0;
-        this->fZ = 0;
-    };
+    }
 
-    CVector(float fX, float fY, float fZ)
+    constexpr CVector(const CVector4D& vec) noexcept :
+        fX(vec.fX),
+        fY(vec.fY),
+        fZ(vec.fZ)
     {
-        this->fX = fX;
-        this->fY = fY;
-        this->fZ = fZ;
     }
 
-    CVector(const CVector4D& vec)
+    constexpr CVector& operator=(const CVector4D& vec) noexcept
     {
-        this->fX = vec.fX;
-        this->fY = vec.fY;
-        this->fZ = vec.fZ;
+        fX = vec.fX;
+        fY = vec.fY;
+        fZ = vec.fZ;
+        return *this;
     }
 
+
+    constexpr CVector Clone() const { return *this; }
+
+    // returns the length of the vector and normalizes it
     float Normalize()
     {
-        float t = sqrt(fX * fX + fY * fY + fZ * fZ);
+        const float t = Length();
         if (t > FLOAT_EPSILON)
         {
-            float fRcpt = 1 / t;
-            fX *= fRcpt;
-            fY *= fRcpt;
-            fZ *= fRcpt;
+            fX /= t;
+            fY /= t;
+            fZ /= t;
+
+            return t;
         }
         else
-            t = 0;
-        return t;
+            return 0;
     }
 
-    float Length() const { return sqrt((fX * fX) + (fY * fY) + (fZ * fZ)); }
+    inline float Length() const { return sqrt((fX * fX) + (fY * fY) + (fZ * fZ)); }
 
-    float LengthSquared() const { return (fX * fX) + (fY * fY) + (fZ * fZ); }
+    // LengthSquared returns Length() without sqrt applied (i.e. returns x*x* + y*y + z*z)
+    // For more info see CMaterialLine3DBatcher::Flush()
```

Comment:
I don't think it's necessary to include the second line here because it's common to use `LengthSquared` for comparisons, which is when you don't strictly care about the square root.

If you still believe it's worth mentioning, then it might be clearer to have a comment like this, which is more descriptive and doesn't require the user to dig into potentially unrelated code.

```suggestion
    // LengthSquared returns Length() without sqrt applied (i.e. returns x*x* + y*y + z*z)
    // This can be useful if you only want to compare lengths.
```